File: trunk/ZendAMF/app/servicos/ServicosGrupoProduto.php
<?php

include 'app/vo/GrupoProdutoVO.php';

class ServicosGrupoProduto {
	
	
	private $conn;
	
	function ServicosGrupoProduto() {
		$db = new BaseDados();
		$this->conn = $db->conn;
	}
	
	public function addGrupoProduto(GrupoProdutoVO $grupoproduto){
		
		$sql = "insert into grupoproduto (descricao) values ('{$grupoproduto->descricao}')";
		
		$resultado = $this->conn->Execute($sql);
		$grupoproduto->codigo = $this->conn->insert_Id();
		return $grupoproduto;
		
	}
	
	public function removerGrupoProduto(GrupoProdutoVO $grupoproduto){
		
		$sql = "delete from grupoproduto  where codigo = {$grupoproduto->codigo}";
		$resultado = $this->conn->Execute($sql);
		if($this->conn->Affected_Rows()==0){
			return false;
		}
		return true;
	}
	
	public function pesquisarGrupoProduto($texto,$coluna){
		$sql = "select * from grupoproduto where $coluna like '%$texto%'";
		$resultado = $this->conn->Execute($sql);
		while($registro = $resultado->FetchNextObject()){			
			$dados_grupoproduto = new GrupoProdutoVO();
			$dados_grupoproduto->codigo = $registro->CODIGO;
			$dados_grupoproduto->descricao = $registro->DESCRICAO;
			$retorna_dados_grupoproduto [] = $dados_grupoproduto;
		}
		return $retorna_dados_grupoproduto;
	}
	
	public function getGrupoProdutos(){
		$sql = "select * from grupoproduto";
		
		$resultado = $this->conn->Execute($sql);
		
		while($registro = $resultado->FetchNextObject()){			
			$dados_grupoproduto = new GrupoProdutoVO();
			$dados_grupoproduto->codigo = $registro->CODIGO;
			$dados_grupoproduto->descricao = $registro->DESCRICAO;
			$retorna_dados_grupoproduto [] = $dados_grupoproduto;
		}
		return $retorna_dados_grupoproduto;	
	}
}


//$eu = new ServicosGrupoProduto();
//$grupoproduto = new GrupoProdutoVO();
//$grupoproduto->codigo = 2;
//echo $eu->removerGrupoProduto($grupoproduto);

?>

File: ServidorJava/WEB-INF/src/services/ServicosGrupoProduto.java
package services;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import vo.GrupoProdutoVO;

public class ServicosGrupoProduto {

	public GrupoProdutoVO addGrupoProduto(GrupoProdutoVO grupoproduto) {

		String sql = "insert into grupoproduto (descricao) values ('"
				+ grupoproduto.descricao + "')";

		Statement st;
		this.banco.conectar();
		this.banco.getConexao().setAutoCommit(false);
		st = this.banco.getConexao().createStatement();
		if (st.executeUpdate(sql, Statement.RETURN_GENERATED_KEYS) == 0) {
			throw new RuntimeException("Erro ao salvar grupo de produto!");
		}
		ResultSet rs = st.getGeneratedKeys();
		if (rs.next())
			grupoproduto.codigo = rs.getInt(1);
		else
			throw new RuntimeException("Erro ao salvar grupo de produto!");
		this.banco.getConexao().commit();
		return grupoproduto;

	}

	public boolean removerGrupoProduto(GrupoProdutoVO grupoproduto) {
		String sql = "delete from grupoproduto  where codigo = "
				+ grupoproduto.codigo + "";
		if (banco.executarNoQuery(sql) == 0) {
			return false;
		}
		return true;
	}

	public GrupoProdutoVO atualizarGrupoProduto(GrupoProdutoVO grupoproduto) {
		String sql = "UPDATE grupoproduto SET descricao = '"
				+ grupoproduto.descricao + "' where codigo = "
				+ grupoproduto.codigo + "";
		if (banco.executarNoQuery(sql) == 0) {
			throw new RuntimeException("Erro ao atualizar grupo produto!");
		}
		return grupoproduto;

	}

	public ArrayList<GrupoProdutoVO> pesquisarGrupoProduto(String texto,
			String coluna) {
		String sql = "select * from grupoproduto where " + coluna + " like '%"
				+ texto + "%'";
		ResultSet rs = banco.executar(sql);
		return this.toGrupoProduto(rs);
	}

	public ArrayList<GrupoProdutoVO> getGrupoProdutos() {
		String sql = "select * from grupoproduto";
		ResultSet rs = banco.executar(sql);
		return this.toGrupoProduto(rs);
	}

	public boolean isExiste(GrupoProdutoVO grupoproduto) {
		String sql = "select * from grupoproduto where codigo = '"+grupoproduto.codigo+"' OR descricao = '"
				+ grupoproduto.descricao + "'";
		ResultSet rs;
		rs = this.banco.executar(sql);
		if (rs.next())
			return true;
		else
			return false;

	}

	private ArrayList<GrupoProdutoVO> toGrupoProduto(ResultSet rs)
			throws SQLException {
		ArrayList<GrupoProdutoVO> gp = new ArrayList<GrupoProdutoVO>();
		while (rs.next()) {
			GrupoProdutoVO dados_item = new GrupoProdutoVO();
			dados_item.codigo = rs.getInt("codigo");
			dados_item.descricao = rs.getString("descricao");

			gp.add(dados_item);
		}
		return gp;
	}

}
